robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowContext.java

```diff
@@ -58,15 +55,12 @@ public Resources getResources() {
 
   @Implementation
   public File getExternalCacheDir() {
-    EXTERNAL_CACHE_DIR.mkdirs();
-    return EXTERNAL_CACHE_DIR;
+    return ShadowEnvironment.getExternalStorageDirectory();
   }
 
   @Implementation
   public File getExternalFilesDir(String type) {
-    File f = (type == null) ? EXTERNAL_FILES_DIR : new File( EXTERNAL_FILES_DIR, type );
-    f.mkdirs();
-    return f;
+    return ShadowEnvironment.getExternalStoragePublicDirectory(type);
```

Comment:
Shouldn't we prefer to call the Android API methods here, e.g:

Environment.getExternalStoragePublicDirectory() ?

That way if we ever decided to change the way we shadow that method we wouldn't need to refactor this class.